diff --git BUILD.boost BUILD.boost
index a27b588..f33d733 100644
--- BUILD.boost
+++ BUILD.boost
@@ -531,6 +531,10 @@ boost_library(
     deps = [
         ":config",
         ":detail",
+        ":type_traits",
+        ":utility",
+        ":smart_ptr",
+        ":shared_ptr",
     ],
 )
 
@@ -1434,7 +1438,6 @@ boost_library(
     deps = [
         ":current_function",
         ":detail",
-        ":exception",
     ],
 )
 
@@ -1466,6 +1469,7 @@ boost_library(
         ":system",
         ":tuple",
         ":type_traits",
+        ":exception",
     ],
 )
 
@@ -1746,6 +1750,8 @@ boost_library(
     ":type_traits",
     ":core",
     ":config",
+    ":proto",
+    ":implicit_cast",
   ]
 )
 
@@ -1757,15 +1763,27 @@ boost_library(
     ":config",
     ":lexical_cast",
     ":function",
+    ":intrusive_ptr",
   ]
 )
 
+boost_library(
+    name = "pending",
+)
+
 boost_library(
   name = "graph",
   deps = [
+    ":unordered",
     ":property_map",
     ":property_tree",
     ":lexical_cast",
     ":xpressive",
+    ":pending",
+    ":typeof",
+    ":tuple",
+    ":fusion",
+    ":algorithm",
+    ":parameter",
   ]
 )
