diff --git BUILD.boost BUILD.boost
index a27b588..f33d733 100644
--- BUILD.boost
+++ BUILD.boost
@@ -527,10 +527,14 @@ boost_library(
     name = "exception",
     hdrs = [
         "boost/exception_ptr.hpp",
+        "boost/shared_ptr/shared_ptr.hpp",
     ],
     deps = [
         ":config",
         ":detail",
+        ":type_traits",
+        ":utility",
+        ":shared_ptr",
     ],
 )
 
@@ -1746,6 +1750,8 @@ boost_library(
     ":type_traits",
     ":core",
     ":config",
+    ":proto",
+    ":implicit_cast",
   ]
 )
 
@@ -1757,15 +1763,27 @@ boost_library(
     ":config",
     ":lexical_cast",
     ":function",
+    ":intrusive_ptr",
   ]
 )
 
+boost_library(
+    name = "pending",
+)
+
 boost_library(
   name = "graph",
   deps = [
+    ":unordered",
     ":property_map",
     ":property_tree",
     ":lexical_cast",
     ":xpressive",
+    ":pending",
+    ":typeof",
+    ":tuple",
+    ":fusion",
+    ":algorithm",
+    ":parameter",
   ]
 )

