diff --git BUILD.boost BUILD.boost
index a27b588..f33d733 100644
--- BUILD.boost
+++ BUILD.boost
@@ -531,6 +531,7 @@ boost_library(
     deps = [
         ":config",
         ":detail",
+        ":type_traits",
     ],
 )
 
@@ -1736,6 +1737,10 @@ boost_library(
 )
 
 boost_library(
+    name = "pending",
+)
+
+boost_library(
     name = "winapi",
     visibility = ["//visibility:private"],
 )
@@ -1746,6 +1751,8 @@ boost_library(
     ":type_traits",
     ":core",
     ":config",
+    ":proto",
+    ":implicit_cast",
   ]
 )
 
@@ -1757,15 +1764,23 @@ boost_library(
     ":config",
     ":lexical_cast",
     ":function",
+    ":intrusive_ptr",
   ]
 )
 
 boost_library(
   name = "graph",
   deps = [
+    "unordered",
     ":property_map",
     ":property_tree",
     ":lexical_cast",
     ":xpressive",
+    ":pending",
+    ":typeof",
+    ":tuple",
+    ":fusion",
+    ":algorithm",
+    ":parameter",
   ]
 )
