diff --git BUILD.boost BUILD.boost
index a27b588..f33d733 100644
--- BUILD.boost
+++ BUILD.boost
@@ -531,6 +531,7 @@ boost_library(
     deps = [
         ":config",
         ":detail",
+        ":type_traits",
     ],
 )
 
@@ -1741,6 +1742,10 @@ boost_library(
 )
 
 boost_library(
+    name = "pending",
+)
+
+boost_library(
   name = "xpressive",
   deps = [
     ":type_traits",
@@ -1757,15 +1762,19 @@ boost_library(
     ":config",
     ":lexical_cast",
     ":function",
+    ":intrusive_ptr",
   ]
 )
 
 boost_library(
   name = "graph",
   deps = [
+    ":unordered",
     ":property_map",
     ":property_tree",
     ":lexical_cast",
     ":xpressive",
+    ":pending",
+    ":typeof",
   ]
 )
