diff --git BUILD.boost BUILD.boost
index fd03694..0ff42aa 100644
--- BUILD.boost
+++ BUILD.boost
@@ -1739,3 +1739,49 @@ boost_library(
     name = "winapi",
     visibility = ["//visibility:private"],
 )
+
+boost_library(
+    name = "unit_test_framework",
+    srcs = [
+        "libs/test/src/compiler_log_formatter.cpp",
+        "libs/test/src/debug.cpp",
+        "libs/test/src/decorator.cpp",
+        "libs/test/src/execution_monitor.cpp",
+        "libs/test/src/framework.cpp",
+        "libs/test/src/plain_report_formatter.cpp",
+        "libs/test/src/progress_monitor.cpp",
+        "libs/test/src/results_collector.cpp",
+        "libs/test/src/results_reporter.cpp",
+        "libs/test/src/test_framework_init_observer.cpp",
+        "libs/test/src/test_tools.cpp",
+        "libs/test/src/test_tree.cpp",
+        "libs/test/src/unit_test_log.cpp",
+        "libs/test/src/unit_test_main.cpp",
+        "libs/test/src/unit_test_monitor.cpp",
+        "libs/test/src/unit_test_parameters.cpp",
+        "libs/test/src/junit_log_formatter.cpp",
+        "libs/test/src/xml_log_formatter.cpp",
+        "libs/test/src/xml_report_formatter.cpp",
+    ],
+    deps = [":test"],
+)
+
+boost_library(
+  name = "xpressive",
+  deps = [
+    ":type_traits",
+    ":core",
+    ":config",
+  ]
+)
+
+boost_library(
+  name = "property_map",
+  deps = [
+    ":type_traits",
+    ":core",
+    ":config",
+    ":lexical_cast",
+    ":function",
+  ]
+)
