diff --git a/config.h b/config.h
new file mode 100644
index 00000000..6bb7fb5d
--- /dev/null
+++ b/config.h
@@ -0,0 +1,146 @@
+/* config.h.  Generated from config.h.in by configure.  */
+/* config.h.in.  Generated from configure.ac by autoheader.  */
+
+/* dconf is 0.13.4 or later */
+#define DCONF_0_13_4 TRUE
+
+/* Define to 1 if translation of program messages to the user's native
+   language is requested. */
+#define ENABLE_NLS 1
+
+/* Enable key snooper */
+#define ENABLE_SNOOPER TRUE
+
+/* Enable surrounding-text support */
+#define ENABLE_SURROUNDING TRUE
+
+/* The prefix for out gettext translation domains. */
+#define GETTEXT_PACKAGE "ibus10"
+
+/* Define to 1 if you have the Mac OS X function CFLocaleCopyCurrent in the
+   CoreFoundation framework. */
+/* #undef HAVE_CFLOCALECOPYCURRENT */
+
+/* Define to 1 if you have the Mac OS X function CFPreferencesCopyAppValue in
+   the CoreFoundation framework. */
+/* #undef HAVE_CFPREFERENCESCOPYAPPVALUE */
+
+/* Define to 1 if you have the `daemon' function. */
+#define HAVE_DAEMON 1
+
+/* Define if the GNU dcgettext() function is already present or preinstalled.
+   */
+#define HAVE_DCGETTEXT 1
+
+/* Define to 1 if you have the <dlfcn.h> header file. */
+#define HAVE_DLFCN_H 1
+
+/* Define if the GNU gettext() function is already present or preinstalled. */
+#define HAVE_GETTEXT 1
+
+/* Define if you have the iconv() function and it works. */
+/* #undef HAVE_ICONV */
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#define HAVE_INTTYPES_H 1
+
+/* Define to 1 if you have the <memory.h> header file. */
+#define HAVE_MEMORY_H 1
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#define HAVE_STDINT_H 1
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#define HAVE_STDLIB_H 1
+
+/* Define to 1 if you have the <strings.h> header file. */
+#define HAVE_STRINGS_H 1
+
+/* Define to 1 if you have the <string.h> header file. */
+#define HAVE_STRING_H 1
+
+/* Define to 1 if you have the <sys/prctl.h> header file. */
+#define HAVE_SYS_PRCTL_H 1
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#define HAVE_SYS_STAT_H 1
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#define HAVE_SYS_TYPES_H 1
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#define HAVE_UNISTD_H 1
+
+/* Define to 1 if you have the <X11/XKBlib.h> header file. */
+#define HAVE_X11_XKBLIB_H 1
+
+/* The default socket directory to connect ibus-daemon. */
+#define IBUS_SOCKET_DIR "unix:tmpdir=$XDG_CACHE_HOME/ibus"
+
+/* Define the location where the catalogs will be installed */
+#define LOCALEDIR "/usr/local/share/locale"
+
+/* Define to the sub-directory where libtool stores uninstalled libraries. */
+#define LT_OBJDIR ".libs/"
+
+/* Does not enbale keyboard snooper in those applications */
+#define NO_SNOOPER_APPS "firefox.*,.*chrome.*,.*chromium.*,terminator"
+
+/* Name of package */
+#define PACKAGE "ibus"
+
+/* Define to the address where bug reports for this package should be sent. */
+#define PACKAGE_BUGREPORT "https://github.com/ibus/ibus/issues"
+
+/* Define to the full name of this package. */
+#define PACKAGE_NAME "ibus"
+
+/* Define to the full name and version of this package. */
+#define PACKAGE_STRING "ibus 1.5.22"
+
+/* Define to the one symbol short name of this package. */
+#define PACKAGE_TARNAME "ibus"
+
+/* Define to the home page for this package. */
+#define PACKAGE_URL ""
+
+/* Define to the version of this package. */
+#define PACKAGE_VERSION "1.5.22"
+
+/* Define to 1 if you have the ANSI C header files. */
+#define STDC_HEADERS 1
+
+/* Enable extensions on AIX 3, Interix.  */
+#ifndef _ALL_SOURCE
+# define _ALL_SOURCE 1
+#endif
+/* Enable GNU extensions on systems that have them.  */
+#ifndef _GNU_SOURCE
+# define _GNU_SOURCE 1
+#endif
+/* Enable threading extensions on Solaris.  */
+#ifndef _POSIX_PTHREAD_SEMANTICS
+# define _POSIX_PTHREAD_SEMANTICS 1
+#endif
+/* Enable extensions on HP NonStop.  */
+#ifndef _TANDEM_SOURCE
+# define _TANDEM_SOURCE 1
+#endif
+/* Enable general extensions on Solaris.  */
+#ifndef __EXTENSIONS__
+# define __EXTENSIONS__ 1
+#endif
+
+
+/* Version number of package */
+#define VERSION "1.5.22"
+
+/* Define to 1 if on MINIX. */
+/* #undef _MINIX */
+
+/* Define to 2 if the system does not provide POSIX.1 features except with
+   this defined. */
+/* #undef _POSIX_1_SOURCE */
+
+/* Define to 1 if you need to in order for `stat' and other things to work. */
+/* #undef _POSIX_SOURCE */
diff --git a/src/ibusversion.h b/src/ibusversion.h
new file mode 100644
index 00000000..f25dcaee
--- /dev/null
+++ b/src/ibusversion.h
@@ -0,0 +1,72 @@
+/* vim:set et sts=4: */
+/* ibus - The Input Bus
+ * Copyright (C) 2008-2010 Peng Huang <shawn.p.huang@gmail.com>
+ * Copyright (C) 2008-2010 Red Hat, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#ifndef __IBUS_VERSION_H_
+#define __IBUS_VERSION_H_
+
+/**
+ * SECTION: ibusversion
+ * @short_description: Current version of IBus.
+ * @stability: Stable
+ *
+ * IBusVersion shows the current IBus version.
+ */
+
+/* compile time version
+ */
+/**
+ * IBUS_MAJOR_VERSION:
+ *
+ * IBus major version.
+ */
+#define IBUS_MAJOR_VERSION				(1)
+
+/**
+ * IBUS_MINOR_VERSION:
+ *
+ * IBus minor version.
+ */
+#define IBUS_MINOR_VERSION				(5)
+
+/**
+ * IBUS_MICRO_VERSION:
+ *
+ * IBus micro version.
+ */
+#define IBUS_MICRO_VERSION				(22)
+
+/**
+ * IBUS_CHECK_VERSION:
+ * @major: Major version of ibus.
+ * @minor: Minor version of ibus.
+ * @micro: Micro version of ibus.
+ *
+ * Check whether the current IBus version is equal to or greater than
+ * given major.minor.micro.
+ */
+#define	IBUS_CHECK_VERSION(major,minor,micro)	\
+    (IBUS_MAJOR_VERSION > (major) || \
+     (IBUS_MAJOR_VERSION == (major) && IBUS_MINOR_VERSION > (minor)) || \
+     (IBUS_MAJOR_VERSION == (major) && IBUS_MINOR_VERSION == (minor) && \
+      IBUS_MICRO_VERSION >= (micro)))
+
+#endif
+
