diff --git a/freeglut/freeglut/src/fg_version.h b/freeglut/freeglut/src/fg_version.h
index 20bed2f..f26aa34 100644
--- a/freeglut/freeglut/src/fg_version.h
+++ b/freeglut/freeglut/src/fg_version.h
@@ -37,11 +37,11 @@
 #endif
 
 #ifndef VERSION_MINOR
-#define VERSION_MINOR 0
+#define VERSION_MINOR 2
 #endif
 
 #ifndef VERSION_PATCH
-#define VERSION_PATCH 0
+#define VERSION_PATCH 1
 #endif
 
 #endif
