diff --git a/toolchain/BUILD.llvm_repo b/toolchain/BUILD.llvm_repo
index 2f80dab..f79e224 100644
--- a/toolchain/BUILD.llvm_repo
+++ b/toolchain/BUILD.llvm_repo
@@ -14,41 +14,15 @@
 
 package(default_visibility = ["//visibility:public"])
 
-# Some targets may need to directly depend on these files.
-exports_files(glob([
-    "bin/*",
-    "lib/*",
-]))
 
-## LLVM toolchain files
-
-filegroup(
-    name = "clang",
-    srcs = [
-        "bin/clang",
-        "bin/clang++",
-        "bin/clang-cpp",
-    ],
-)
-
-filegroup(
-    name = "ld",
-    srcs = [
-        "bin/ld.lld",
-    ],
-)
-
-filegroup(
-    name = "include",
-    srcs = glob([
+cc_library(
+    name = "hdrs",
+    hdrs = glob([
         "include/**/c++/**",
+        "include/clang-c/**",
         "lib/clang/*/include/**",
     ]),
-)
-
-filegroup(
-    name = "bin",
-    srcs = glob(["bin/**"]),
+    includes = ["include"]
 )
 
 filegroup(
@@ -72,67 +46,4 @@ filegroup(
         ],
         allow_empty = True
     ),
-)
-
-filegroup(
-    name = "ar",
-    srcs = ["bin/llvm-ar"],
-)
-
-filegroup(
-    name = "as",
-    srcs = [
-        "bin/clang",
-        "bin/llvm-as",
-    ],
-)
-
-filegroup(
-    name = "nm",
-    srcs = ["bin/llvm-nm"],
-)
-
-filegroup(
-    name = "objcopy",
-    srcs = ["bin/llvm-objcopy"],
-)
-
-filegroup(
-    name = "objdump",
-    srcs = ["bin/llvm-objdump"],
-)
-
-filegroup(
-    name = "profdata",
-    srcs = ["bin/llvm-profdata"],
-)
-
-filegroup(
-    name = "dwp",
-    srcs = ["bin/llvm-dwp"],
-)
-
-filegroup(
-    name = "ranlib",
-    srcs = ["bin/llvm-ranlib"],
-)
-
-filegroup(
-    name = "readelf",
-    srcs = ["bin/llvm-readelf"],
-)
-
-filegroup(
-    name = "strip",
-    srcs = ["bin/llvm-strip"],
-)
-
-filegroup(
-    name = "symbolizer",
-    srcs = ["bin/llvm-symbolizer"],
-)
-
-filegroup(
-    name = "clang-tidy",
-    srcs = ["bin/clang-tidy"],
-)
+)
\ No newline at end of file
