diff --git a/libtiff/tif_config.h b/libtiff/tif_config.h
index f838623d..7514d5e5 100644
--- a/libtiff/tif_config.h
+++ b/libtiff/tif_config.h
@@ -1,16 +1,36 @@
-/*
-    This file is used on systems not using configure only and contains
-    hardcoded definitions for MSVC, Xcode and MinGW when using makefiles.
- */
+/* libtiff/tif_config.h.  Generated from tif_config.h.in by configure.  */
+/* libtiff/tif_config.h.in.  Generated from configure.ac by autoheader.  */
 
-#ifdef __APPLE__
+/* Define if building universal (internal helper macro) */
+/* #undef AC_APPLE_UNIVERSAL_BUILD */
+
+/* Support CCITT Group 3 & 4 algorithms */
+#define CCITT_SUPPORT 1
+
+/* Pick up YCbCr subsampling info from the JPEG data stream to support files
+   lacking the tag (default enabled). */
+#define CHECK_JPEG_YCBCR_SUBSAMPLING 1
+
+/* enable partial strip reading for large strips (experimental) */
+/* #undef CHUNKY_STRIP_READ_SUPPORT */
+
+/* Support C++ stream API (requires C++ compiler) */
+#define CXX_SUPPORT 1
+
+/* Treat extra sample as alpha (default enabled). The RGBA interface will
+   treat a fourth sample with no EXTRASAMPLE_ value as being ASSOCALPHA. Many
+   packages produce RGBA files but don't mark the alpha properly. */
+#define DEFAULT_EXTRASAMPLE_AS_ALPHA 1
+
+/* enable deferred strip/tile offset/size loading */
+/* #undef DEFER_STRILE_LOAD */
 
 /* Define to 1 if you have the <assert.h> header file. */
 #define HAVE_ASSERT_H 1
 
 /* Define to 1 if you have the declaration of `optarg', and to 0 if you don't.
    */
-#undef HAVE_DECL_OPTARG
+#define HAVE_DECL_OPTARG 1
 
 /* Define to 1 if you have the <dlfcn.h> header file. */
 #define HAVE_DLFCN_H 1
@@ -18,9 +38,6 @@
 /* Define to 1 if you have the <fcntl.h> header file. */
 #define HAVE_FCNTL_H 1
 
-/* Define to 1 if you have the `floor' function. */
-#define HAVE_FLOOR 1
-
 /* Define to 1 if fseeko (and presumably ftello) exists and is declared. */
 #define HAVE_FSEEKO 1
 
@@ -28,7 +45,7 @@
 #define HAVE_GETOPT 1
 
 /* Define to 1 if you have the <GLUT/glut.h> header file. */
-#define HAVE_GLUT_GLUT_H 1
+/* #undef HAVE_GLUT_GLUT_H */
 
 /* Define to 1 if you have the <GL/glut.h> header file. */
 #define HAVE_GL_GLUT_H 1
@@ -43,56 +60,29 @@
    machine */
 #define HAVE_IEEEFP 1
 
-/* Define to 1 if the system has the type `int16'. */
-/* #undef HAVE_INT16 */
-
-/* Define to 1 if the system has the type `int32'. */
-/* #undef HAVE_INT32 */
-
-/* Define to 1 if the system has the type `int8'. */
-/* #undef HAVE_INT8 */
-
 /* Define to 1 if you have the <inttypes.h> header file. */
 #define HAVE_INTTYPES_H 1
 
 /* Define to 1 if you have the <io.h> header file. */
-/* #undef HAVE_IO_H */
-
-/* Define to 1 if you have the `isascii' function. */
-#define HAVE_ISASCII 1
+#define HAVE_IO_H 1
 
 /* Define to 1 if you have the `jbg_newlen' function. */
-/* #undef HAVE_JBG_NEWLEN */
+#define HAVE_JBG_NEWLEN 1
 
 /* Define to 1 if you have the `lfind' function. */
 #define HAVE_LFIND 1
 
-/* Define to 1 if you have the <limits.h> header file. */
-#define HAVE_LIMITS_H 1
-
-/* Define to 1 if you have the <malloc.h> header file. */
-/* #undef HAVE_MALLOC_H */
-
-/* Define to 1 if you have the `memmove' function. */
-#define HAVE_MEMMOVE 1
-
 /* Define to 1 if you have the <memory.h> header file. */
 #define HAVE_MEMORY_H 1
 
-/* Define to 1 if you have the `memset' function. */
-#define HAVE_MEMSET 1
-
 /* Define to 1 if you have the `mmap' function. */
 #define HAVE_MMAP 1
 
 /* Define to 1 if you have the <OpenGL/glu.h> header file. */
-#define HAVE_OPENGL_GLU_H 1
+/* #undef HAVE_OPENGL_GLU_H */
 
 /* Define to 1 if you have the <OpenGL/gl.h> header file. */
-#define HAVE_OPENGL_GL_H 1
-
-/* Define to 1 if you have the `pow' function. */
-#define HAVE_POW 1
+/* #undef HAVE_OPENGL_GL_H */
 
 /* Define if you have POSIX threads libraries and header files. */
 #define HAVE_PTHREAD 1
@@ -101,14 +91,11 @@
 #define HAVE_SEARCH_H 1
 
 /* Define to 1 if you have the `setmode' function. */
-#define HAVE_SETMODE 1
+/* #undef HAVE_SETMODE */
 
 /* Define to 1 if you have the `snprintf' function. */
 #define HAVE_SNPRINTF 1
 
-/* Define to 1 if you have the `sqrt' function. */
-#define HAVE_SQRT 1
-
 /* Define to 1 if you have the <stdint.h> header file. */
 #define HAVE_STDINT_H 1
 
@@ -118,24 +105,18 @@
 /* Define to 1 if you have the `strcasecmp' function. */
 #define HAVE_STRCASECMP 1
 
-/* Define to 1 if you have the `strchr' function. */
-#define HAVE_STRCHR 1
-
 /* Define to 1 if you have the <strings.h> header file. */
 #define HAVE_STRINGS_H 1
 
 /* Define to 1 if you have the <string.h> header file. */
 #define HAVE_STRING_H 1
 
-/* Define to 1 if you have the `strrchr' function. */
-#define HAVE_STRRCHR 1
-
-/* Define to 1 if you have the `strstr' function. */
-#define HAVE_STRSTR 1
-
 /* Define to 1 if you have the `strtol' function. */
 #define HAVE_STRTOL 1
 
+/* Define to 1 if you have the `strtoll' function. */
+#define HAVE_STRTOLL 1
+
 /* Define to 1 if you have the `strtoul' function. */
 #define HAVE_STRTOUL 1
 
@@ -145,9 +126,6 @@
 /* Define to 1 if you have the <sys/stat.h> header file. */
 #define HAVE_SYS_STAT_H 1
 
-/* Define to 1 if you have the <sys/time.h> header file. */
-#define HAVE_SYS_TIME_H 1
-
 /* Define to 1 if you have the <sys/types.h> header file. */
 #define HAVE_SYS_TYPES_H 1
 
@@ -160,38 +138,21 @@
 /* Define to 1 if you have the <windows.h> header file. */
 /* #undef HAVE_WINDOWS_H */
 
-#ifdef __BIG_ENDIAN__
-    /* Set the native cpu bit order */
-    #define HOST_FILLORDER FILLORDER_MSB2LSB
-    
-    /* Define to 1 if your processor stores words with the most significant byte
-    first (like Motorola and SPARC, unlike Intel and VAX). */
-    #define WORDS_BIGENDIAN 1
-#else
-    /* Set the native cpu bit order */
-    #define HOST_FILLORDER FILLORDER_LSB2MSB
-    
-    /* Define to 1 if your processor stores words with the most significant byte
-        first (like Motorola and SPARC, unlike Intel and VAX). */
-    /* #undef WORDS_BIGENDIAN */
-#endif
+/* Native cpu byte order: 1 if big-endian (Motorola) or 0 if little-endian
+   (Intel) */
+#define HOST_BIGENDIAN 0
 
-/* Define to `__inline__' or `__inline' if that's what the C compiler
-   calls it, or to nothing if 'inline' is not supported under any name.  */
-#ifndef __cplusplus
-# ifndef inline
-#  define inline __inline
-# endif
-#endif
+/* Set the native cpu bit order (FILLORDER_LSB2MSB or FILLORDER_MSB2LSB) */
+#define HOST_FILLORDER FILLORDER_LSB2MSB
 
 /* Support ISO JBIG compression (requires JBIG-KIT library) */
-/* #undef JBIG_SUPPORT */
+#define JBIG_SUPPORT 1
 
 /* 8/12 bit libjpeg dual mode enabled */
 /* #undef JPEG_DUAL_MODE_8_12 */
 
 /* Support JPEG compression (requires IJG JPEG library) */
-/* #undef JPEG_SUPPORT */
+#define JPEG_SUPPORT 1
 
 /* 12bit libjpeg primary include file with path */
 /* #undef LIBJPEG_12_PATH */
@@ -203,7 +164,7 @@
 #define LT_OBJDIR ".libs/"
 
 /* Support LZMA2 compression */
-/* #undef LZMA_SUPPORT */
+#define LZMA_SUPPORT 1
 
 /* Support LZW algorithm */
 #define LZW_SUPPORT 1
@@ -215,7 +176,28 @@
 #define NEXT_SUPPORT 1
 
 /* Support Old JPEG compresson (read-only) */
-/* #undef OJPEG_SUPPORT */
+#define OJPEG_SUPPORT 1
+
+/* Name of package */
+#define PACKAGE "tiff"
+
+/* Define to the address where bug reports for this package should be sent. */
+#define PACKAGE_BUGREPORT "tiff@lists.maptools.org"
+
+/* Define to the full name of this package. */
+#define PACKAGE_NAME "LibTIFF Software"
+
+/* Define to the full name and version of this package. */
+#define PACKAGE_STRING "LibTIFF Software 4.1.0"
+
+/* Define to the one symbol short name of this package. */
+#define PACKAGE_TARNAME "tiff"
+
+/* Define to the home page for this package. */
+#define PACKAGE_URL ""
+
+/* Define to the version of this package. */
+#define PACKAGE_VERSION "4.1.0"
 
 /* Support Macintosh PackBits algorithm */
 #define PACKBITS_SUPPORT 1
@@ -231,32 +213,26 @@
 #define SIZEOF_SIGNED_INT 4
 
 /* The size of `signed long', as computed by sizeof. */
-#define SIZEOF_SIGNED_LONG 4
+#define SIZEOF_SIGNED_LONG 8
 
 /* The size of `signed long long', as computed by sizeof. */
 #define SIZEOF_SIGNED_LONG_LONG 8
 
-/* The size of `signed short', as computed by sizeof. */
-#define SIZEOF_SIGNED_SHORT 2
-
 /* The size of `size_t', as computed by sizeof. */
 #define SIZEOF_SIZE_T 8
 
 /* The size of `unsigned char *', as computed by sizeof. */
-#define SIZEOF_UNSIGNED_CHAR_P 4
+#define SIZEOF_UNSIGNED_CHAR_P 8
 
 /* The size of `unsigned int', as computed by sizeof. */
 #define SIZEOF_UNSIGNED_INT 4
 
 /* The size of `unsigned long', as computed by sizeof. */
-#define SIZEOF_UNSIGNED_LONG 4
+#define SIZEOF_UNSIGNED_LONG 8
 
 /* The size of `unsigned long long', as computed by sizeof. */
 #define SIZEOF_UNSIGNED_LONG_LONG 8
 
-/* The size of `unsigned short', as computed by sizeof. */
-#define SIZEOF_UNSIGNED_SHORT 2
-
 /* Define to 1 if you have the ANSI C header files. */
 #define STDC_HEADERS 1
 
@@ -283,10 +259,10 @@
 #define TIFF_INT32_T signed int
 
 /* Signed 64-bit type formatter */
-#define TIFF_INT64_FORMAT "%lld"
+#define TIFF_INT64_FORMAT "%ld"
 
 /* Signed 64-bit type */
-#define TIFF_INT64_T signed long long
+#define TIFF_INT64_T signed long
 
 /* Signed 8-bit type */
 #define TIFF_INT8_T signed char
@@ -298,10 +274,10 @@
 #define TIFF_PTRDIFF_T ptrdiff_t
 
 /* Size type formatter */
-#define TIFF_SIZE_FORMAT "%zd"
+#define TIFF_SIZE_FORMAT "%lu"
 
 /* Unsigned size type */
-#define TIFF_SIZE_T size_t
+#define TIFF_SIZE_T unsigned long
 
 /* Signed size type formatter */
 #define TIFF_SSIZE_FORMAT "%ld"
@@ -319,10 +295,10 @@
 #define TIFF_UINT32_T unsigned int
 
 /* Unsigned 64-bit type formatter */
-#define TIFF_UINT64_FORMAT "%llu"
+#define TIFF_UINT64_FORMAT "%lu"
 
 /* Unsigned 64-bit type */
-#define TIFF_UINT64_T unsigned long long
+#define TIFF_UINT64_T unsigned long
 
 /* Unsigned 8-bit type */
 #define TIFF_UINT8_T unsigned char
@@ -330,113 +306,64 @@
 /* Define to 1 if you can safely include both <sys/time.h> and <time.h>. */
 #define TIME_WITH_SYS_TIME 1
 
-#else /* __APPLE__ */
+/* Define to 1 if your <sys/time.h> declares `struct tm'. */
+/* #undef TM_IN_SYS_TIME */
 
-/* Define to 1 if you have the <assert.h> header file. */
-#define HAVE_ASSERT_H 1
+/* define to use win32 IO system */
+/* #undef USE_WIN32_FILEIO */
 
-/* Define to 1 if you have the <fcntl.h> header file. */
-#define HAVE_FCNTL_H 1
+/* Version number of package */
+#define VERSION "4.1.0"
 
-/* Define as 0 or 1 according to the floating point format suported by the
-   machine */
-#define HAVE_IEEEFP 1
-
-/* Define to 1 if you have the `jbg_newlen' function. */
-#define HAVE_JBG_NEWLEN 1
-
-/* Define to 1 if you have the <string.h> header file. */
-#define HAVE_STRING_H 1
-
-/* Define to 1 if you have the <sys/types.h> header file. */
-#define HAVE_SYS_TYPES_H 1
-
-/* Define to 1 if you have the <io.h> header file. */
-#define HAVE_IO_H 1
-
-/* Define to 1 if you have the <search.h> header file. */
-#define HAVE_SEARCH_H 1
+/* Support webp compression */
+#define WEBP_SUPPORT 1
 
-/* Define to 1 if you have the `setmode' function. */
-#define HAVE_SETMODE 1
-
-/* Define to 1 if you have the `snprintf' function. */
-#define HAVE_SNPRINTF 1
-
-/* The size of a `int', as computed by sizeof. */
-#define SIZEOF_INT 4
-
-/* The size of a `long', as computed by sizeof. */
-#define SIZEOF_LONG 4
-
-/* Signed 64-bit type formatter */
-#define TIFF_INT64_FORMAT "%I64d"
+/* Define WORDS_BIGENDIAN to 1 if your processor stores words with the most
+   significant byte first (like Motorola and SPARC, unlike Intel). */
+#if defined AC_APPLE_UNIVERSAL_BUILD
+# if defined __BIG_ENDIAN__
+#  define WORDS_BIGENDIAN 1
+# endif
+#else
+# ifndef WORDS_BIGENDIAN
+/* #  undef WORDS_BIGENDIAN */
+# endif
+#endif
 
-/* Signed 64-bit type */
-#define TIFF_INT64_T signed __int64
+/* Define to 1 if the X Window System is missing or not being used. */
+/* #undef X_DISPLAY_MISSING */
 
-/* Unsigned 64-bit type formatter */
-#define TIFF_UINT64_FORMAT "%I64u"
+/* Support Deflate compression */
+#define ZIP_SUPPORT 1
 
-/* Unsigned 64-bit type */
-#define TIFF_UINT64_T unsigned __int64
+/* Support zstd compression */
+/* #undef ZSTD_SUPPORT */
 
-/* Set the native cpu bit order */
-#define HOST_FILLORDER FILLORDER_LSB2MSB
+/* Enable large inode numbers on Mac OS X 10.5.  */
+#ifndef _DARWIN_USE_64_BIT_INODE
+# define _DARWIN_USE_64_BIT_INODE 1
+#endif
 
-/*
-    Use _snprintf() with older versions of MSVC and MinGW.
+/* Number of bits in a file offset, on hosts where this is settable. */
+/* #undef _FILE_OFFSET_BITS */
 
-    Note that we can't do this unconditionally as starting from the version
-    which does have it (MSVS 2015 a.k.a. MSVC 14 a.k.a. _MSC_VER 19.00), it
-    doesn't allow redefining snprintf any longer.
+/* Define to 1 to make fseeko visible on some hosts (e.g. glibc 2.2). */
+/* #undef _LARGEFILE_SOURCE */
 
-    Also, MinGW-w32 6.3 uses macro-hackery in its stdio.h which breaks if it is
-    redefined so, again, only do this for earlier versions.
- */
-#if (defined(_MSC_VER) && _MSC_VER < 1900) || \
-    (defined(__MINGW32__) && __GNUC__ < 6)
-# define snprintf _snprintf
-#endif
+/* Define for large files, on AIX-style hosts. */
+/* #undef _LARGE_FILES */
 
-/* Define to 1 if your processor stores words with the most significant byte
-   first (like Motorola and SPARC, unlike Intel and VAX). */
-/* #undef WORDS_BIGENDIAN */
+/* Define to empty if `const' does not conform to ANSI C. */
+/* #undef const */
 
 /* Define to `__inline__' or `__inline' if that's what the C compiler
    calls it, or to nothing if 'inline' is not supported under any name.  */
 #ifndef __cplusplus
-# ifndef inline
-#  define inline __inline
-# endif
+/* #undef inline */
 #endif
 
-#if !defined (__BORLANDC__) && !defined (__WATCOMC__)
-   #define lfind _lfind
-#endif
+/* Define to `long int' if <sys/types.h> does not define. */
+/* #undef off_t */
 
-#ifdef _WIN32_WCE
-#   undef HAVE_FCNTL_H
-#   undef HAVE_SYS_TYPES_H
-
-    /*
-       CE headers don't define these standard constants (not even underscored
-       versions), provide our own replacements as they seem to be only used in
-       libtiff own code anyhow.
-     */
-#   define   O_RDONLY    0x0000
-#   define   O_WRONLY    0x0001
-#   define   O_RDWR      0x0002
-#   define   O_CREAT     0x0100
-#   define   O_TRUNC     0x0200
-#   define   O_EXCL      0x0400
-#endif /* _WIN32_WCE */
-
-#endif /* __APPLE__/!__APPLE__ */
-/*
- * Local Variables:
- * mode: c
- * c-basic-offset: 8
- * fill-column: 78
- * End:
- */
+/* Define to `unsigned int' if <sys/types.h> does not define. */
+/* #undef size_t */
diff --git a/libtiff/tiffconf.h b/libtiff/tiffconf.h
index f0dc9bb0..12159d89 100644
--- a/libtiff/tiffconf.h
+++ b/libtiff/tiffconf.h
@@ -1,11 +1,4 @@
-/*
-    This file is used on systems not using configure only and contains
-    hardcoded definitions for MSVC, Xcode and MinGW when using makefiles.
-
-    As an additional twist, the settings that ought to be in this file are in
-    tif_config.h when using Xcode (__APPLE__ defined).
- */
-
+/* libtiff/tiffconf.h.  Generated from tiffconf.h.in by configure.  */
 /*
   Configuration defines for installed libtiff.
   This file maintained for backward compatibility. Do not use definitions
@@ -15,78 +8,35 @@
 #ifndef _TIFFCONF_
 #define _TIFFCONF_
 
-#ifdef __APPLE__
-#include "tif_config.h"
-#else
-
-#ifdef __MINGW32__
-#include <basetyps.h>
-#endif
-
-/* Define to 1 if the system has the type `int16'. */
-/* #undef HAVE_INT16 */
-
-/* Define to 1 if the system has the type `int32'. */
-/* #undef HAVE_INT32 */
+/* Signed 16-bit type */
+#define TIFF_INT16_T signed short
 
-/* Define to 1 if the system has the type `int8'. */
-/* #undef HAVE_INT8 */
+/* Signed 32-bit type */
+#define TIFF_INT32_T signed int
 
-/* The size of a `int', as computed by sizeof. */
-#define SIZEOF_INT 4
+/* Signed 64-bit type */
+#define TIFF_INT64_T signed long
 
 /* Signed 8-bit type */
 #define TIFF_INT8_T signed char
 
-/* Unsigned 8-bit type */
-#define TIFF_UINT8_T unsigned char
-
-/* Signed 16-bit type */
-#define TIFF_INT16_T signed short
-
 /* Unsigned 16-bit type */
 #define TIFF_UINT16_T unsigned short
 
-/* Signed 32-bit type formatter */
-#define TIFF_INT32_FORMAT "%d"
-
-/* Signed 32-bit type */
-#define TIFF_INT32_T signed int
-
-/* Unsigned 32-bit type formatter */
-#define TIFF_UINT32_FORMAT "%u"
-
 /* Unsigned 32-bit type */
 #define TIFF_UINT32_T unsigned int
 
-/* Signed 64-bit type formatter */
-#define TIFF_INT64_FORMAT "%I64d"
-
-/* Signed 64-bit type */
-#define TIFF_INT64_T signed __int64
-
-/* Unsigned 64-bit type formatter */
-#define TIFF_UINT64_FORMAT "%I64u"
-
 /* Unsigned 64-bit type */
-#define TIFF_UINT64_T unsigned __int64
+#define TIFF_UINT64_T unsigned long
+
+/* Unsigned 8-bit type */
+#define TIFF_UINT8_T unsigned char
 
 /* Signed size type */
-#if defined(_WIN64)
-#define TIFF_SSIZE_T signed __int64
-#else
-#define TIFF_SSIZE_T signed int
-#endif
-
-/* Signed size type formatter */
-#if defined(_WIN64)
-#define TIFF_SSIZE_FORMAT "%I64d"
-#else
-#define TIFF_SSIZE_FORMAT "%ld"
-#endif
+#define TIFF_SSIZE_T signed long
 
 /* Pointer difference type */
-#define TIFF_PTRDIFF_T long
+#define TIFF_PTRDIFF_T ptrdiff_t
 
 /* Compatibility stuff. */
 
@@ -107,6 +57,9 @@
 /* Support JPEG compression (requires IJG JPEG library) */
 #define JPEG_SUPPORT 1
 
+/* Support JBIG compression (requires JBIG-KIT library) */
+#define JBIG_SUPPORT 1
+
 /* Support LogLuv high dynamic range encoding */
 #define LOGLUV_SUPPORT 1
 
@@ -118,7 +71,7 @@
 
 /* Support Old JPEG compresson (read contrib/ojpeg/README first! Compilation
    fails with unpatched IJG JPEG library) */
-/* #undef OJPEG_SUPPORT */
+#define OJPEG_SUPPORT 1
 
 /* Support Macintosh PackBits algorithm */
 #define PACKBITS_SUPPORT 1
@@ -163,12 +116,4 @@
 #define PHOTOSHOP_SUPPORT
 #define IPTC_SUPPORT
 
-#endif /* __APPLE__/!__APPLE__ */
 #endif /* _TIFFCONF_ */
-/*
- * Local Variables:
- * mode: c
- * c-basic-offset: 8
- * fill-column: 78
- * End:
- */
