diff --git BUILD.boost BUILD.boost
index 8bf78c1..031b4d1 100644
--- BUILD.boost
+++ BUILD.boost
@@ -1359,6 +1359,7 @@ boost_library(
         ":function",
         ":operators",
         ":type_traits",
+        ":smart_ptr",
     ],
 )
 
@@ -1381,6 +1382,8 @@ boost_library(
         ":shared_ptr",
         ":throw_exception",
         ":utility",
+        ":cstdint",
+        ":tuple",
     ],
 )
 
